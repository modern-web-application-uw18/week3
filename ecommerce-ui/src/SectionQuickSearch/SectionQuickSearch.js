import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Card from '../Card/Card.js';

export default class QuickSearches extends Component {

    render() {
      const quickSearch = this.props.quickSearch;
      const searches = quickSearch.map((search, index) => {
        return <Card search={search} title={search.title} image={search.image} key={index} index={index}/>
      });

        return (
          <section className="searches">
            <h2>Quick Searches</h2>
            {searches}
          </section>
        );
    }
}

QuickSearches.propTypes = {
  'quickSearch': PropTypes.array.isRequired
}

QuickSearches.defaultProps = {
  quickSearch: [{
    title: 'test_title',
    houseAmount: 0,
    image: 'test_image',
    description: 'test_description'
  }]
}
