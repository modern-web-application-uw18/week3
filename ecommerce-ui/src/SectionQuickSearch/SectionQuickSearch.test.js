import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';

import SectionQuickSearch from './SectionQuickSearch.js';

describe('SectionQuickSearch', () => {
  it('Smoke test', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SectionQuickSearch />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Match Snapshot', () => {
    const component = TestRenderer.create(<SectionQuickSearch />);
    expect(component.toJSON()).toMatchSnapshot();
  });

})
