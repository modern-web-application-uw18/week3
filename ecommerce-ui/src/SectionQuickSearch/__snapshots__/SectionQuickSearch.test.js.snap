// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionQuickSearch Match Snapshot 1`] = `
<section
  className="searches"
>
  <h2>
    Quick Searches
  </h2>
  <div
    className="card"
  >
    <div
      className="coverImage"
      style={
        Object {
          "background": "url('test_image') center / cover no-repeat",
        }
      }
    >
      <span
        className="textOverlay"
      >
        test_title
      </span>
    </div>
    <div
      className="cardInfo"
    >
      <div
        className="quickSearch"
      >
        <div
          className="title"
        >
          0
           airbnbs available
        </div>
        <div
          className="text"
        >
          test_description
        </div>
      </div>
    </div>
  </div>
</section>
`;
